hr_expense_invoice/models/hr_expense_sheet.py

```diff
@@ -112,3 +143,30 @@ def action_view_invoices(self):
             action["view_mode"] = "tree,form"
             action["domain"] = [("id", "in", invoice_ids)]
         return action
+
+    def _compute_state(self):
+        """Set proper state according linked invoices."""
+        sheets_with_invoices = self.filtered(
+            lambda sheet: all(
+                expense.invoice_id and expense.invoice_id.state == "posted"
+                for expense in sheet.expense_line_ids
+            )
+            and sheet.state == sheet.approval_state
+        )
+
+        company_account_sheets = sheets_with_invoices.filtered(
+            lambda sheet: sheet.payment_mode == "company_account"
+        )
+        company_account_sheets.write({"state": "done"})
```

Comment:
You should not write inside compute methods

`company_account_sheets.state = "done"`

Same below.

Reply:
done